gensim/models/tfidfmodel.py

```diff
@@ -65,21 +71,43 @@ def resolve_weights(smartirs):
         doesn't fit the list of permissible values.
 
     """
+
+    if not smartirs:
+        return None
+
+    if isinstance(smartirs, str) and re.match(r"...\....", smartirs):
+        match = re.match(r"(?P<ddd>...)\.(?P<qqq>...)", smartirs)
+        raise ValueError(
+            "The notation {ddd}.{qqq} specifies two term-weighting schemes, "
+            "one for collection documents ({ddd}) and one for queries ({qqq}). "
+            "You must train two separate tf-idf models.".format(
+                ddd=match.group("ddd"),
+                qqq=match.group("qqq"),
+            )
+        )
     if not isinstance(smartirs, str) or len(smartirs) != 3:
         raise ValueError("Expected a string of length 3 except got " + smartirs)
 
     w_tf, w_df, w_n = smartirs
 
-    if w_tf not in 'nlabL':
-        raise ValueError("Expected term frequency weight to be one of 'nlabL', except got {}".format(w_tf))
+    if w_tf not in 'btnaldL':
+        raise ValueError("Expected term frequency weight to be one of 'btnaldL', except got {}".format(w_tf))
+
+    if w_df not in 'xnftp':
+        raise ValueError("Expected inverse document frequency weight to be one of 'xnftp', except got {}".format(w_df))
 
-    if w_df not in 'ntp':
-        raise ValueError("Expected inverse document frequency weight to be one of 'ntp', except got {}".format(w_df))
+    if w_n not in 'xncub':
+        raise ValueError("Expected normalization weight to be one of 'xncub', except got {}".format(w_n))
```

Comment:
```suggestion
        raise ValueError("Expected normalization weight to be one of 'xncub', got {}".format(w_n))
```

Reply:
Lang fix: Please drop the word `except` from all these messages.

Reply:
Fixed in 13081f76.